app/models/post.rb

```diff
@@ -1,3 +1,21 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  # Attributes
+  attribute :title, :string
+  attribute :text, :text
+  attribute :comments_counter, :integer, default: 0
+  attribute :likes_counter, :integer, default: 0
+
+  # A method to update the posts counter for a user
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
[OPTIONAL] It is a good practice to use rails callbacks for methods like this. This way you can make the method private thereby improving the security of your backend